pkg/apis/sparkoperator.k8s.io/v1beta2/types.go

```diff
@@ -162,6 +162,19 @@ type ScheduledSparkApplicationList struct {
 	Items           []ScheduledSparkApplication `json:"items,omitempty"`
 }
 
+// Specific SparkUI config parameters
+type UIConfig struct {
+	// Ingressclass defines the class of ingress used for k8s annotation: kubernetes.io/ingressclass
+	// +optional
+	IngressClass *string `json:"ingressClass,omitempty"`
+	// EnableSSL is necessary to specify that k8s ingress Will use SSL
+	// +optional
+	EnableSSL bool `json:"enableSSL,omitempty"`
+	// EnableSSL combined with ForceSSL will set Ingress k8s annotation: nginx.ingress.kubernetes.io/force-ssl-redirect to true
+	// +optional
+	ForceSSL bool `json:"forceSSL,omitempty"`
```

Comment:
Will `forceSSL` ever be false if `enableSSL` is true? For example, if `ingressClass` is `nginx`, is it likely that `forceSSL` is false when `enableSSL` is true? I still don't quite understand the relationship between the two boolean fields. Looking at this more, I tend to think that it might make sense to reorganize `UIConfig` (renamed to `SparkUIIngressConfig`) like the following:

```go
// SparkUIIngressConfig is for Ingress configuration for the Spark driver UI.
type struct SparkUIIngressConfig {
  // Looks like an Ingress must reference an IngressClass, so this should not be an optional field. 
  IngressClass string `json:"ingressClass"`
  Nginx *NginxIngressConfig `json:"nginxConfig,omitempty"`
}

// NginxIngressConfig is for configuring the NGINX Ingress.
type NginxIngressConfig struct {
  EnableSSL bool `json:"enableSSL,omitempty"`
  ForceSSLRedirect bool `json:"forceSSLRedirect,omitempty"`
}
```